### ISSUE TITLE
Drop Python 2.6 compatibility?

### ISSUE THREAD

**Issue by mperrin** [spacetelescope/webbpsf#21]
<a href="https://github.com/josePhoenix"><img src="https://avatars3.githubusercontent.com/u/172886?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [josePhoenix](https://github.com/josePhoenix)**
_Thursday Oct 09, 2014 at 20:00 GMT_
_Originally opened as https://github.com/mperrin/webbpsf/issues/21_

----

It looks like the promise of Python 2.6 compatibility is no longer true for WebbPSF. I'm fine with dropping it, and moving towards a bilingual codebase with Python 3 support (i.e. #2 ).

I notice plhstins1 and the witservs have older Python installs, so it may be necessary to maintain compatibility. If so, we should see if we can configure Travis-CI to test on Python 2.6. Or, at least, write up some instructions for setting up a Python 2.6 testing environment to make sure it's all still working.

**Comment by mperrin** [comment]
<a href="https://github.com/josePhoenix"><img src="https://avatars3.githubusercontent.com/u/172886?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [josePhoenix](https://github.com/josePhoenix)**
_Thursday Oct 09, 2014 at 20:02 GMT_

----

After installing Python 2.6.9 through `conda`, and master from josephoenix/poppy and josephoenix/webbpsf, this is the error I was getting:

```
$ python
Python 2.6.9 |Continuum Analytics, Inc.| (unknown, Jan 10 2014, 13:33:57)
[GCC 4.2.1 (Apple Inc. build 5577)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import webbpsf
WebbPSF log messages of level INFO and above will be shown.
WebbPSF log outputs will be directed to the screen.
>>> webbpsf.Instrument('NIRCam')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jlong/src/webbpsf/webbpsf/webbpsf_core.py", line 1288, in Instrument
    if name == 'nircam': return NIRCam()
  File "/Users/jlong/src/webbpsf/webbpsf/webbpsf_core.py", line 871, in __init__
    self.detector=self.detector_list[0]
  File "/Users/jlong/src/webbpsf/webbpsf/webbpsf_core.py", line 262, in detector
    self._detector = DetectorGeometry(self.name, siaf_aperture_name, shortname=detname)
  File "/Users/jlong/src/webbpsf/webbpsf/webbpsf_core.py", line 1343, in __init__
    self.mysiaf = SIAF(instr=self.instrname, basepath=os.path.join( utils.get_webbpsf_data_path(), self.instrname) )
  File "/Users/jlong/src/webbpsf/webbpsf/jwxml.py", line 576, in __init__
    for entry in self._tree.getroot().iter('SiafEntry'):
AttributeError: iter
>>>
```

**Comment by mperrin** [comment]
<a href="https://github.com/mperrin"><img src="https://avatars0.githubusercontent.com/u/1151745?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mperrin](https://github.com/mperrin)**
_Wednesday Nov 05, 2014 at 23:31 GMT_

----

Just a comment here I agree with all of the above, and we should just update the docs to say >= 2.7.

**Comment by mperrin** [comment]
<a href="https://github.com/josePhoenix"><img src="https://avatars3.githubusercontent.com/u/172886?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [josePhoenix](https://github.com/josePhoenix)**
_Wednesday Dec 10, 2014 at 15:59 GMT_

----

Closed by #27
